builtin/providers/azurerm/resource_arm_virtual_machine_scale_set.go

```diff
@@ -735,19 +763,29 @@ func flattenAzureRMVirtualMachineScaleSetOsProfile(profile *compute.VirtualMachi
 
 func flattenAzureRmVirtualMachineScaleSetStorageProfileOSDisk(profile *compute.VirtualMachineScaleSetOSDisk) []interface{} {
 	result := make(map[string]interface{})
-	result["name"] = *profile.Name
+
+	if profile.Name != nil {
+		result["name"] = *profile.Name
+	}
+
 	if profile.Image != nil {
+		result["name"] = *profile.Name
 		result["image"] = *profile.Image.URI
 	}
 
 	if profile.VhdContainers != nil {
+		result["name"] = *profile.Name
```

Comment:
given [this statement](https://github.com/hashicorp/terraform/pull/13717/files#diff-376b49ee73ee83c9f4097665b9e249aeR767) exists above, this line can be removed :)